v1/python-sdk/tutorials/automl-with-azureml/forecasting-data-preparation/auto-ml-forecasting-data-preparation.ipynb

```diff
@@ -0,0 +1,980 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "target_column_name = \"usage\"\n",
+    "time_column_name = \"date\"\n",
+    "GRAIN_COL = \"customer_id\"\n",
+    "time_series_id_column_names = [GRAIN_COL]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In the followng block of code we will download the data from the original source to the `data` folder, load the data and print the first few rows."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if not IS_ORIGINAL_DATASET_DOWNLOADED:\n",
+    "    print(\"Downloading data from the source ...\\n---\")\n",
+    "    # Download original data\n",
+    "    from io import BytesIO\n",
+    "    from urllib.request import urlopen\n",
+    "    from zipfile import ZipFile\n",
+    "\n",
+    "    zipurl = \"https://archive.ics.uci.edu/ml/machine-learning-databases/00321/LD2011_2014.txt.zip\"\n",
+    "\n",
+    "    with urlopen(zipurl) as zipresp:\n",
+    "        with ZipFile(BytesIO(zipresp.read())) as zfile:\n",
+    "            zfile.extractall(\"data\")\n",
+    "    IS_ORIGINAL_DATASET_DOWNLOADED = True\n",
+    "\n",
+    "DATA_LOCATION = os.path.join(os.getcwd(), \"data\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(\"Printing the first few rows of the downloaded data ...\\n---\")\n",
+    "raw_df = pd.read_table(\"data/LD2011_2014.txt\", sep=\";\", low_memory=False)\n",
+    "print(raw_df.head())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "raw_df.columns"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The downloaded data is in \"wide\" format, meaning each column name that starts with \"MT_xxx\" represents one unique time series. The first columnn \"Unnamed: 0\" is actually a time stamp at which the obervation for every time series takes place. Let's raname this column to something more meaningful. We will call it `date`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "raw_df.rename(columns={raw_df.columns[0]: \"date\"}, inplace=True)\n",
+    "raw_df.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The \"wide\" data format is not usable in AutoML, which is deisgned to accept a \"long\" format data. To tranform the data from wide to long format, we will take the each uniqe time seres (date, MT_xxx) and stack them vertically. The end result will be a data frame containing 3 columns: (i) the `date` column, (ii) `custmer_id` column, which is the identifier of the time series and is derived from the column name in the wide format, and (iii) `usage` which is the target varaible we are trying to model."
```

Comment:
Can we have brief explanation on what is ths long format as well? Either "in each unique time series, there is only one row for each timestamp", or "the same format that is accepted by typical scikit-learn machine learning models".

Reply:
Thank you. Fixed.